test/Nancy.Tests/Unit/NancyEngineFixture.cs

```diff
@@ -340,6 +340,45 @@ public void HandleRequest_should_get_context_from_context_factory()
         }
 
         [Fact]
+        public async Task Should_catch_exception_inside_status_code_handler_and_return_default_error_response()
+        {
+            // Given
+            A.CallTo(() => this.requestDispatcher.Dispatch(A<NancyContext>.Ignored, A<CancellationToken>.Ignored))
+                .Returns(new Response() { StatusCode = HttpStatusCode.InternalServerError });
+
+            var engine =
+                new NancyEngine(this.requestDispatcher, this.contextFactory, new[] { this.statusCodeHandler, new DefaultStatusCodeHandler(this.negotiator, this.environment), }, A.Fake<IRequestTracing>(), new DisabledStaticContentProvider(), this.negotiator, this.environment)

```

Comment:
I shortened it. this is the only test that has this line, if we have 2+ then I think it'd make sense for a variable